### PR TITLE
Assorted mappings, mainly enchanting, some rendering params and generator type.

### PR DESCRIPTION
**Pull request by i509VCB**

### CODE REVIEW

File: mappings/net/minecraft/enchantment/EnchantmentHelper.mapping [hunk 1]
```diff
@@ -53,14 +58,25 @@ CLASS net/minecraft/class_1890 net/minecraft/enchantment/EnchantmentHelper
...
 		ARG 0 stack
-	METHOD method_8209 forEachEnchantment (Lnet/minecraft/class_1890$class_1891;Ljava/lang/Iterable;)V
-		ARG 0 action
+	METHOD method_8207 (Lnet/minecraft/class_1309;Lnet/minecraft/class_1297;Lnet/minecraft/class_1887;I)V
+		ARG 2 enchantment
+		ARG 3 level
+	METHOD method_8208 (Lorg/apache/commons/lang3/mutable/MutableFloat;Lnet/minecraft/class_1310;Lnet/minecraft/class_1887;I)V
+		ARG 2 enchantment
+		ARG 3 level
+	METHOD method_8209 acceptEnchantments (Lnet/minecraft/class_1890$class_1891;Ljava/lang/Iterable;)V
```
Comment (liach) [inline mappings/net/minecraft/enchantment/EnchantmentHelper.mapping]: Hmm, debatable name

File: mappings/net/minecraft/enchantment/EnchantmentHelper.mapping [hunk 2]
```diff
@@ -156,7 +175,10 @@ CLASS net/minecraft/class_1890 net/minecraft/enchantment/EnchantmentHelper
 			COMMENT whether treasure enchantments may appear
 	METHOD method_8234 getEfficiency (Lnet/minecraft/class_1309;)I
 		ARG 0 entity
-	CLASS class_1891 Consumer
+	CLASS class_1891 Visitor
```
Comment (liach) [inline mappings/net/minecraft/enchantment/EnchantmentHelper.mapping]: :-1: This is just a consumer.

File: mappings/net/minecraft/util/registry/Registry.mapping [hunk 3]
```diff
@@ -148,3 +151,12 @@ CLASS net/minecraft/class_2378 net/minecraft/util/registry/Registry
 		ARG 1 value
 	METHOD method_29722 getEntries ()Ljava/util/Set;
 	METHOD method_30517 getKey ()Lnet/minecraft/class_5321;
+	METHOD method_31138 getLifecycle ()Lcom/mojang/serialization/Lifecycle;
+	METHOD method_31139 getEntryLifecycle (Ljava/lang/Object;)Lcom/mojang/serialization/Lifecycle;
+	METHOD method_31140 require (Lnet/minecraft/class_5321;)Ljava/lang/Object;
```
Comment (liach) [inline mappings/net/minecraft/util/registry/Registry.mapping]: Really need a better name for this.

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/util/registry/Registry.mapping (hunk 3 above)
Comment (shartte) [reply mappings/net/minecraft/util/registry/Registry.mapping]: getRequired?

Comment (i509VCB) [reply mappings/net/minecraft/util/registry/Registry.mapping]: That makes me think of a method which gets a registry entry which is marked as required.

This is basically `get(RegistryKey<T>)` but non-null and throws if the value is not present

Comment (liach) [reply mappings/net/minecraft/util/registry/Registry.mapping]: Apparently cannot think of a better name than `require` unfortunately. I honestly would rename the optional one to `find` but that's out of scope. This is good.

Comment (YanisBft) [reply mappings/net/minecraft/util/registry/Registry.mapping]:
```suggestion
	METHOD method_31140 getRequired (Lnet/minecraft/class_5321;)Ljava/lang/Object;
```
I think `getRequired` is still better than `require`

Comment (Gegy) [reply mappings/net/minecraft/util/registry/Registry.mapping]: `getOrExcept`/`getOrThrow`?

Comment (i509VCB) [reply mappings/net/minecraft/util/registry/Registry.mapping]: getRequired makes me think it has to be a builtin biome to use the method. getOrThrow could work but is not as succinct as `require`

Comment (liach) [reply mappings/net/minecraft/util/registry/Registry.mapping]: `require` is better than `getRequired`

**Review by liach — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Review by Gegy — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**